### PR TITLE
Update deps and countrycode

### PR DESCRIPTION
**Pull request by yehezkielgunawan**
# Description
Update deps and the API url for country code phone call

# Linked Issues
-

# Preview or Screenshot
-

### CODE REVIEW

**Review by lauslim12 — COMMENTED** [review]
Looks good for me, but I have minor questions.

File: src/lib/fetcher/fetcher.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ import axios, { AxiosResponse } from "axios";
 import { BASE_URL, API_KEY } from "@/constants/baseConfig";
 
 export type CountryCodeProps = {
-  name: string;
-  dial_code: string;
-  code: string;
+  E164: string;
+  phone_code: string;
+  country_name: string;
```
[the remark this answers is not in the document]
Comment (yehezkielgunawan) [reply src/lib/fetcher/fetcher.ts:8]: For this case, I followed the data type from the API [https://countrycode.dev/api/calls](https://countrycode.dev/api/calls)

File: src/pages/index.tsx [hunk 2]
```diff
@@ -117,9 +117,9 @@ const Home = ({ countryCodes }: { countryCodes: Array<CountryCodeProps> }) => {
               <datalist id="countryCode">
                 {countryCodeList.map((code, index) => (
                   <option
-                    value={code.dial_code}
+                    value={code.phone_code}
                     key={index}
```
[the remark this answers is not in the document]
Comment (yehezkielgunawan) [reply src/pages/index.tsx:121]: Hm... I see, it's better to use the data ID as the key. I'll push the revised commit soon, then please approve it if there's no other concern😅

File: src/lib/fetcher/fetcher.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lauslim12) [reply src/lib/fetcher/fetcher.ts:8]: Ahh, I see. I usually transform it to `camelCase` via [zod library](https://github.com/colinhacks/zod).

But it's a bit overkill for now, so let's not do that.

**Review by lauslim12 — APPROVED** [review]
LGTM 🚀

**Pull request merged**